Maze.h

```diff
@@ -21,6 +22,9 @@ class Maze {
 	static const unsigned MOVEMENT_COST = 100;
 	static const unsigned MOVEMENT_COST_DIAGONAL = 141;
 
+	static Maze loadMazeFromFile(std::string fileName);
```

Comment:
Perhaps just call it `fromFile`? I think `Maze::fromFile()` should be clear enough.